v2/worker/src/jobs.ts

```diff
@@ -55,122 +57,116 @@ export class Job extends BrigadierJob {
-        const client = http2.connect(
-          this.event.worker.apiAddress,
-          {
+    return new Promise<void>((resolve, reject) => {
+      const eventSource = new EventSource(
+        `${this.event.worker.apiAddress}/v2/events/${this.event.id}/worker/jobs/${this.name}/status?watch=true&sse=true`, 
+        {
+          https: {
             // TODO: Get our hands on the API server's CA to validate the cert
-            rejectUnauthorized: false,
-          }
-        )
-        client.on("error", (err) => console.error(err))
-        req = client.request({
-          ":path": `/v2/events/${this.event.id}/worker/jobs/${this.name}/status?watch=true`,
-          "Authorization": `Bearer ${this.event.worker.apiToken}`
-        })
-        req.setEncoding("utf8")
-
-        req.on("response", (response) => {
-          const status = response[":status"]
-          if (status != 200) {
-            reject(new Error(`Received ${status} when attempting to stream job status`))
-            abortMonitor = true
-            req.destroy()
-          }
-        })
-
-        req.on("data", (data: string) => {
-          try {
-            const status = JSON.parse(data)
-            this.logger.debug(`Job phase is ${status.phase}`)
-            switch (status.phase) {
-            // TODO: Do we still use this phase???
-            case "ABORTED":
-              reject(new Error("Job was aborted"))
-              abortMonitor = true
-              req.destroy()
-              break
-            case "FAILED":
-              reject(new Error("Job failed"))
-              abortMonitor = true
-              req.destroy()
-              break
-            case "SUCCEEDED":
-              resolve()
-              abortMonitor = true
-              req.destroy()
-              break
-            case "TIMED_OUT":
-              reject(new Error("Job timed out"))
-              abortMonitor = true
-              req.destroy()
-            }
-          } catch (e) {
-            // Let it stay connected
-          } 
-        })
-
-        req.on("end", () => {
-          client.destroy()
-          if (!abortMonitor) {
-            // We got disconnected, but apparently not deliberately, so try
-            // again.
-            this.logger.debug("Had to restart the job monitor")
-            startMonitorReq()
+            rejectUnauthorized: false
+          },
+          headers: {
+            "Authorization": `Bearer ${this.event.worker.apiToken}`
           }
-        })
-      }
-      startMonitorReq() // This starts the monitor for the first time.
+        }
+      )
+      eventSource.addEventListener("message", (event: any) => { // eslint-disable-line @typescript-eslint/no-explicit-any
+        let status: any // eslint-disable-line @typescript-eslint/no-explicit-any
+        try {
+          status = JSON.parse(event.data)
+        } catch(e) {
+          eventSource.close() 
+          reject(new Error("Error parsing job status"))
```

Comment:
Do we want to include/log `e` as well?  (Would inform a few other similar spots, if so).

Reply:
Yes... we should probably be _wrapping_ the original exception.

I'll find the other spots where that might be applicable as well.

Reply:
https://github.com/brigadecore/brigade/pull/1217/commits/298693b3d29f4d5477cd03fd089b8f178e92168e looks great - thanks!